gen/definitions/sda_fabric_site.yaml

```diff
@@ -0,0 +1,29 @@
+---
+name: SDA Fabric Site
+rest_endpoint: /dna/intent/api/v1/business/sda/fabric-site
+get_rest_endpoint: /dna/intent/api/v1/sda/fabricSites
+delete_rest_endpoint: /dna/intent/api/v1/sda/fabricSites
+get_from_all: true
+id_from_query_path: response
+no_update: true
```

Comment:
Instead of doing nothing when an attribute changes, wouldn't it be better to set all attributes to "requires_replace" to re-create the resource with the new attributes?

Reply:
when i change site_name_hierarchy object is being recreated (I had to add id: true to siteNameHierarchy attribute, without this go was throwing errors) so requires_replace is not needed:

```
following symbols:
-/+ destroy and then create replacement

Terraform will perform the following actions:

  # catalystcenter_sda_fabric_site.example must be replaced
-/+ resource "catalystcenter_sda_fabric_site" "example" {
      ~ id                  = "cdbeb9a9-6f1c-4d94-9df9-8b011afe7434" -> (known after apply)
      ~ site_name_hierarchy = "Global/Site2" -> "Global/TestSite" # forces replacement
        # (1 unchanged attribute hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.
```
second parameter can be either FABRIC_SITE or FABRIC_ZONE, and conversion from FABRIC_SITE to FABRIC_ZONE cannot really be done unless parent fabric site exists and only then fabric_zone can be created, thats why i choose not to put requires_replace, moreover i was following logic used in resource for dnacenter provider which only has create, read and delete operations.  Let me know what do you think ?


```
  # catalystcenter_sda_fabric_site.example must be replaced
-/+ resource "catalystcenter_sda_fabric_site" "example" {
      + fabric_type         = "FABRIC_ZONE" # forces replacement
      ~ id                  = "0b7bb7a7-95ea-4ef3-b0ad-8883c4a01cd1" -> (known after apply)
        # (1 unchanged attribute hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

catalystcenter_sda_fabric_site.example: Destroying... [id=0b7bb7a7-95ea-4ef3-b0ad-8883c4a01cd1]
catalystcenter_sda_fabric_site.example: Destruction complete after 4s
catalystcenter_sda_fabric_site.example: Creating...
╷
│ Error: Client Error
│ 
│   with catalystcenter_sda_fabric_site.example,
│   on main.tf line 17, in resource "catalystcenter_sda_fabric_site" "example":
│   17: resource "catalystcenter_sda_fabric_site" "example" {
│ 
│ Failed to configure object (POST), got error: HTTP Request failed: StatusCode 400, {
│   "status" : "failed",
│   "description" : "Parent fabric site must exist before creating Fabric zone under it. Create Fabric site first and then try fabric zone creation. Provide a valid siteNameHierarchy and try again.",
│   "taskId" : null,
│   "taskStatusUrl" : "/dna/intent/api/v1/task/null",
│   "executionStatusUrl" : "/dna/intent/api/v1/dnacaap/management/execution-status/d3124e5b-f4d5-4d18-ae22-405afabecf77",
│   "executionId" : "d3124e5b-f4d5-4d18-ae22-405afabecf77"
```